CMakeLists.txt

```diff
@@ -65,13 +69,65 @@ if(AMALGAMATE_SOURCES)
   endif(BUILD_HEADER_ONLY)
   set(INSTALL_HEADERS ${CMAKE_BINARY_DIR}/amalgamation/miniz.h)
 else(AMALGAMATE_SOURCES)
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```

Comment:
I think having explicit __declspec(dllexport)/__attribute__ ((visibility ("default"))) would be better. Could you do that in a follow up?

Reply:
Yes, in fact I have it half done.
I did not wanted a huge scope of change here in the header files to be part of the first PR.